File: app/src/main/java/ani/saito/downloadAddon/DownloadBuilder.java
package ani.saito.downloadAddon;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Map;
import kotlin.Metadata;
import kotlin.Pair;
import kotlin.collections.CollectionsKt;
import kotlin.jvm.internal.Intrinsics;
import kotlin.text.StringsKt;

@Metadata(d1 = {"\u0000$\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0002\u0010\u000e\n\u0002\b\t\n\u0002\u0010$\n\u0002\b\u0003\u0018\u00002\u00020\u0001B\u0005¢\u0006\u0002\u0010\u0002J\u0006\u0010\u000b\u001a\u00020\u0006J\u0006\u0010\f\u001a\u00020\u0006J \u0010\r\u001a\u00020\u00002\u0018\u0010\u0003\u001a\u0014\u0012\u0010\u0012\u000e\u0012\u0004\u0012\u00020\u0006\u0012\u0004\u0012\u00020\u00060\u00050\u0004J\u000e\u0010\u000e\u001a\u00020\u00002\u0006\u0010\u0007\u001a\u00020\u0006J\u001a\u0010\u000f\u001a\u00020\u00002\u0012\u0010\b\u001a\u000e\u0012\u0004\u0012\u00020\u0006\u0012\u0004\u0012\u00020\u00060\u0010J \u0010\u0011\u001a\u00020\u00002\u0018\u0010\t\u001a\u0014\u0012\u0010\u0012\u000e\u0012\u0004\u0012\u00020\u0006\u0012\u0004\u0012\u00020\u00060\u00050\u0004J\u000e\u0010\u0012\u001a\u00020\u00002\u0006\u0010\n\u001a\u00020\u0006R \u0010\u0003\u001a\u0014\u0012\u0010\u0012\u000e\u0012\u0004\u0012\u00020\u0006\u0012\u0004\u0012\u00020\u00060\u00050\u0004X\u000e¢\u0006\u0002\n\u0000R\u000e\u0010\u0007\u001a\u00020\u0006X\u000e¢\u0006\u0002\n\u0000R\u000e\u0010\b\u001a\u00020\u0006X\u000e¢\u0006\u0002\n\u0000R \u0010\t\u001a\u0014\u0012\u0010\u0012\u000e\u0012\u0004\u0012\u00020\u0006\u0012\u0004\u0012\u00020\u00060\u00050\u0004X\u000e¢\u0006\u0002\n\u0000R\u000e\u0010\n\u001a\u00020\u0006X\u000e¢\u0006\u0002\n\u0000¨\u0006\u0013"}, d2 = {"Lani/dantotsu/downloadAddon/DownloadBuilder;", "", "()V", "audioLinks", "", "Lkotlin/Pair;", "", "downloadPath", "headers", "subtitleLinks", "videoLink", "buildDownload", "buildProbe", "setAudioLinks", "setDownloadPath", "setHeaders", "", "setSubtitleLinks", "setVideoLink", "app_release"}, k = 1, mv = {1, 9, 0}, xi = 48)
/* compiled from: DownloadBuilder.kt */
public final class DownloadBuilder {
    private List<Pair<String, String>> audioLinks = CollectionsKt.emptyList();
    private String downloadPath = "";
    private String headers = "";
    private List<Pair<String, String>> subtitleLinks = CollectionsKt.emptyList();
    private String videoLink = "";

    public final DownloadBuilder setHeaders(Map<String, String> map) {
        Intrinsics.checkNotNullParameter(map, "headers");
        if (!map.isEmpty()) {
            StringBuilder sb = new StringBuilder();
            for (Map.Entry next : map.entrySet()) {
                sb.append("\"" + ((String) next.getKey()) + ": " + ((String) next.getValue()) + "\"'\r\n'");
            }
            String sb2 = sb.toString();
            Intrinsics.checkNotNullExpressionValue(sb2, "toString(...)");
            this.headers = sb2;
            return this;
        }
        throw new IllegalArgumentException("Headers cannot be empty");
    }

    public final DownloadBuilder setAudioLinks(List<Pair<String, String>> list) {
        Intrinsics.checkNotNullParameter(list, "audioLinks");
        Collection arrayList = new ArrayList();
        for (Object next : list) {
            Pair pair = (Pair) next;
            if (!Intrinsics.areEqual(pair.getFirst(), (Object) this.videoLink) && ((CharSequence) pair.getFirst()).length() > 0) {
                arrayList.add(next);
            }
        }
        this.audioLinks = (List) arrayList;
        return this;
    }

    public final DownloadBuilder setSubtitleLinks(List<Pair<String, String>> list) {
        Intrinsics.checkNotNullParameter(list, "subtitleLinks");
        Collection arrayList = new ArrayList();
        for (Object next : list) {
            Pair pair = (Pair) next;
            if (!Intrinsics.areEqual(pair.getFirst(), (Object) this.videoLink) && ((CharSequence) pair.getFirst()).length() > 0) {
                arrayList.add(next);
            }
        }
        this.subtitleLinks = (List) arrayList;
        return this;
    }

    public final DownloadBuilder setVideoLink(String str) {
        Intrinsics.checkNotNullParameter(str, "videoLink");
        if (str.length() != 0) {
            this.videoLink = str;
            return this;
        }
        throw new IllegalArgumentException("Video link cannot be empty");
    }

    public final DownloadBuilder setDownloadPath(String str) {
        Intrinsics.checkNotNullParameter(str, "downloadPath");
        if (str.length() != 0) {
            this.downloadPath = str;
            return this;
        }
        throw new IllegalArgumentException("Download path cannot be empty");
    }

    public final String buildDownload() {
        if (this.downloadPath.length() == 0 || this.videoLink.length() == 0) {
            throw new IllegalArgumentException("Download path and video link cannot be empty");
        }
        int i = 0;
        boolean z = StringsKt.contains((CharSequence) this.videoLink, (CharSequence) ".m3u", false) || StringsKt.contains((CharSequence) this.videoLink, (CharSequence) ".mpd", false);
        StringBuilder sb = new StringBuilder();
        if (this.headers.length() > 0) {
            sb.append("-headers " + this.headers + ' ');
        }
        sb.append("-fflags +discardcorrupt -err_detect ignore_err -reconnect 1 -reconnect_streamed 1 -reconnect_on_network_error 1 -reconnect_on_http_error 1 -reconnect_delay_max 600 -tls_verify 0 ");
        if (z) {
            sb.append("-max_reload 1000 -seg_max_retry 1000 -m3u8_hold_counters 10000 ");
        }
        sb.append("-i \"" + this.videoLink + "\" ");
        for (Pair first : this.audioLinks) {
            sb.append("-i \"" + ((String) first.getFirst()) + "\" ");
        }
        int i2 = 0;
        for (Object next : this.subtitleLinks) {
            int i3 = i2 + 1;
            if (i2 < 0) {
                CollectionsKt.throwIndexOverflow();
            }
            sb.append("-i \"" + ((String) ((Pair) next).getFirst()) + "\" ");
            i2 = i3;
        }
        sb.append("-bufsize 1024k -map 0:v -c:v copy -map 0:a? -c:a copy -map 0:s? ");
        int i4 = 0;
        for (Object next2 : this.audioLinks) {
            int i5 = i4 + 1;
            if (i4 < 0) {
                CollectionsKt.throwIndexOverflow();
            }
            Pair pair = (Pair) next2;
            sb.append("-map " + i5 + ":a ");
            if (((CharSequence) pair.getSecond()).length() > 0) {
                sb.append("-metadata:s:a:" + i4 + " language=" + ((String) pair.getSecond()) + ' ');
            }
            i4 = i5;
        }
        int i6 = 0;
        for (Object next3 : this.subtitleLinks) {
            int i7 = i6 + 1;
            if (i6 < 0) {
                CollectionsKt.throwIndexOverflow();
            }
            Pair pair2 = (Pair) next3;
            sb.append("-map " + (i + this.audioLinks.size() + i7) + ":s -c:s mov_text ");
            if (((CharSequence) pair2.getSecond()).length() > 0) {
                sb.append("-metadata:s:s:" + i6 + " language=" + ((String) pair2.getSecond()) + ' ');
            }
            i6 = i7;
        }
        sb.append(this.downloadPath);
        String sb2 = sb.toString();
        Intrinsics.checkNotNullExpressionValue(sb2, "toString(...)");
        return sb2;
    }

    public final String buildProbe() {
        String str = this.videoLink;
        Intrinsics.checkNotNullExpressionValue(str, "videoLink");
        return "-probesize 10M -analyzeduration 10M -fflags +discardcorrupt -err_detect ignore_err -reconnect 1 -reconnect_streamed 1 -reconnect_on_network_error 1 -reconnect_on_http_error 1 -reconnect_delay_max 600 -tls_verify 0 -i \"" + str + "\" -print_format json -show_format -show_streams";
    }
}

File: app/src/main/java/ani/saito/downloadAddon/DownloadBuilder.kt
package ani.saito.downloadAddon

import kotlin.collections.ArrayList
import kotlin.collections.Map
import kotlin.collections.List
import kotlin.collections.MutableList
import kotlin.collections.emptyList
import kotlin.text.StringBuilder

class DownloadBuilder {

    private var audioLinks: List<Pair<String, String>> = emptyList()
    private var downloadPath: String = ""
    private var headers: String = ""
    private var subtitleLinks: List<Pair<String, String>> = emptyList()
    private var videoLink: String = ""

    fun setHeaders(map: Map<String, String>): DownloadBuilder {
        require(map.isNotEmpty()) { "Headers cannot be empty" }

        val sb = StringBuilder()
        for ((key, value) in map) {
            sb.append("\"$key: $value\"\r\n")
        }
        headers = sb.toString()
        return this
    }

    fun setAudioLinks(list: List<Pair<String, String>>): DownloadBuilder {
        val filteredList = list.filter {
            it.first != videoLink && it.first.isNotEmpty()
        }
        audioLinks = filteredList
        return this
    }

    fun setSubtitleLinks(list: List<Pair<String, String>>): DownloadBuilder {
        val filteredList = list.filter {
            it.first != videoLink && it.first.isNotEmpty()
        }
        subtitleLinks = filteredList
        return this
    }

    fun setVideoLink(str: String): DownloadBuilder {
        require(str.isNotEmpty()) { "Video link cannot be empty" }
        videoLink = str
        return this
    }

    fun setDownloadPath(str: String): DownloadBuilder {
        require(str.isNotEmpty()) { "Download path cannot be empty" }
        downloadPath = str
        return this
    }

    fun buildDownload(): String {
        require(downloadPath.isNotEmpty() && videoLink.isNotEmpty()) { "Download path and video link cannot be empty" }

        val sb = StringBuilder()
        if (headers.isNotEmpty()) {
            sb.append("-headers $headers ")
        }
        sb.append("-fflags +discardcorrupt -err_detect ignore_err -reconnect 1 -reconnect_streamed 1 -reconnect_on_network_error 1 -reconnect_on_http_error 1 -reconnect_delay_max 600 -tls_verify 0 ")

        val isPlaylist = videoLink.contains(".m3u", ignoreCase = true) || videoLink.contains(".mpd", ignoreCase = true)
        if (isPlaylist) {
            sb.append("-max_reload 1000 -seg_max_retry 1000 -m3u8_hold_counters 10000 ")
        }
        sb.append("-i \"$videoLink\" ")

        audioLinks.forEach {
            sb.append("-i \"${it.first}\" ")
        }

        subtitleLinks.forEach {
            sb.append("-i \"${it.first}\" ")
        }

        sb.append("-bufsize 1024k -map 0:v -c:v copy -map 0:a? -c:a copy -map 0:s? ")

        audioLinks.forEachIndexed { index, pair ->
            sb.append("-map ${index + 1}:a ")
            if (pair.second.isNotEmpty()) {
                sb.append("-metadata:s:a:$index language=${pair.second} ")
            }
        }

        subtitleLinks.forEachIndexed { index, pair ->
            sb.append("-map ${audioLinks.size + index + 1}:s ")
            if (pair.second.isNotEmpty()) {
                sb.append("-metadata:s:s:$index language=${pair.second} ")
                sb.append("-metadata:s:s:$index title=${pair.second} ")
            }
        }

        sb.append("-c:s srt -f matroska ")
        sb.append("$downloadPath -v trace")
        return sb.toString()
    }

    fun buildProbe(): String {
        require(videoLink.isNotEmpty()) { "Video link cannot be empty" }

        val sb = StringBuilder()
        if (headers.isNotEmpty()) {
            sb.append("-headers $headers")
        }
        sb.append(" -i \"$videoLink\"")
        sb.append(" -show_entries format=duration -v quiet -of csv=\"p=0\"")
        return sb.toString()
    }
}
